src/plasma/plasma_manager.cc

```diff
@@ -1269,9 +1275,22 @@ void log_object_hash_mismatch_error_result_callback(ObjectID object_id,
                                                     void *user_context) {
   CHECK(!task_id.is_nil());
   PlasmaManagerState *state = (PlasmaManagerState *) user_context;
-  /* Get the specification for the nondeterministic task. */
+/* Get the specification for the nondeterministic task. */
+#if !RAY_USE_NEW_GCS
   task_table_get_task(state->db, task_id, NULL,
                       log_object_hash_mismatch_error_task_callback, state);
+#else
+  RAY_CHECK_OK(state->gcs_client.task_table().Lookup(
+      ray::JobID::nil(), task_id,
+      [user_context](gcs::AsyncGcsClient *client, const TaskID &id,
+                     std::shared_ptr<TaskTableDataT> t) {
+        Task *task = Task_alloc(
```

Comment:
I still found these callbacks not too readable :-) . How about for the unused params, just do 
```c++
[user_context](gcs::AsyncGcsClient*, const TaskID&,
                      std::shared_ptr<TaskTableDataT> t) { ... }
```
or 
```c++
[user_context](gcs::AsyncGcsClient* unused_client, const TaskID& unused_id,
                      std::shared_ptr<TaskTableDataT> t) { ... }
```

Reply:
+1